### PR TITLE
[IMP]: Check Quantity of stock moves while validate the MO

### PR DESCRIPTION
**Pull request by rso-odoo**
Description of the issue/feature this PR addresses:  action_confirm of MO shouldn't validate the SM at zero

Task ID: 2159374

Current behavior before PR: 

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_move.py [hunk 1]
```diff
@@ -1000,7 +1000,7 @@ def _action_confirm(self, merge=True, merge_into=False):
             if move.move_orig_ids:
                 move_waiting |= move
             else:
-                if move.procure_method == 'make_to_order':
+                if move.procure_method == 'make_to_order' and move.product_uom_qty > 0:
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py]: float_compare

File: addons/stock/models/stock_move.py [hunk 2]
```diff
@@ -1181,6 +1181,8 @@ def _action_assign(self):
             else:
                 if not move.move_orig_ids:
                     if move.procure_method == 'make_to_order':
+                        if move.product_uom_qty == 0:
```
Comment (amoyaux) [inline addons/stock/models/stock_move.py]: float_is_zero

**Comment by amoyaux** [comment]
Also should be added in replenish wizard. For skip 0.0

**Comment by sle-odoo** [comment]
replaced by https://github.com/odoo/odoo/pull/45499

**Pull request closed without merging**